index.html

```diff
@@ -1115,6 +1129,36 @@
             ctx.fillRect(0, urlBarOffset, urlBarWidth, urlBarHeight);
             ctx.fillStyle = '#FFF';
             ctx.fillRect(5, urlBarOffset + 5, urlBarWidth - 10, urlBarHeight - 10);
+
+            // Chris`s Loading Bar
+              if(loadingBarState.ifLoading === true && loadingBarState.ifError === false){
+                  ctx.fillStyle = '#333';
+                  ctx.fillRect(0, loadingBarOffset, loadingBarWidth, loadingBarHeight);
+                  ctx.fillStyle = '#FFF';
+                  ctx.fillRect(10, loadingBarOffset, loadingBarWidth -20, loadingBarHeight- 10);
+                  ctx.fillStyle = '#333';
+                  ctx.font = `${fontSize}px Arial`;
+                  ctx.fillText("                Loading...", 10, loadingBarOffset + loadingBarHeight - 10*2 - 30);
+                  if(loadingBarState.ifLoaded === true){
+                      ctx.fillStyle = '#333';
+                      ctx.fillRect(0, loadingBarOffset, loadingBarWidth, loadingBarHeight);
+                      ctx.fillStyle = '#FFF';
+                      ctx.fillRect(10, loadingBarOffset, loadingBarWidth -20, loadingBarHeight- 10);
+                      ctx.fillStyle = '#333';
+                      ctx.font = `${fontSize}px Arial`;
+                      ctx.fillText("                Mesh is loaded!", 10, loadingBarOffset + loadingBarHeight - 10*2 - 30);
+                  }
+              }
+              else if(loadingBarState.ifLoading === true && loadingBarState.ifError === true){
+                  ctx.fillStyle = '#ff1818';
+                  ctx.fillRect(0, loadingBarOffset, loadingBarWidth, loadingBarHeight);
+                  ctx.fillStyle = '#FFF';
+                  ctx.fillRect(10, loadingBarOffset, loadingBarWidth -20, loadingBarHeight- 10);
+                  ctx.fillStyle = '#333';
+                  ctx.font = `${fontSize}px Arial`;
+                  ctx.fillText('          Error mesh could not load!', 10, loadingBarOffset + loadingBarHeight - 10*2 - 30);
```

Comment:
Instead of spaces, this seems like the kind of thing to crunch with the offset math.